axlearn/common/inference.py

```diff
@@ -370,21 +382,15 @@ def _inference_iter(
         **kwargs,
     ) -> Tuple[Tensor, NestedTensor, NestedTensor]:
         """Implements inference for a single input batch."""
-        cfg = self.config
         new_prng_key, iter_key = jax.random.split(prng_key)
 
-        def inference_cast(in_tree: NestedTensor) -> NestedTensor:
```

Comment:
instead of casting params for each inference run (also copies the tensor that needs more memory)